File: CS/03_Images/ConvertAllPagesToPNG.cs
using System;
using System.Drawing;
using System.Drawing.Imaging;
using System.Windows.Forms;
using Spire.Pdf;


namespace ConvertAllPagesToPNG
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            String file = @"..\..\..\..\..\..\Data\ToImage.pdf";

            //Open pdf document
            PdfDocument pdf = new PdfDocument();
            pdf.LoadFromFile(file);

            // Iterate through each page
            for (int i = 0; i < pdf.Pages.Count; i++)
            {
                String fileName = String.Format("ToPNG-img-{0}.png", i);

                //Save page to images in PNG type
                using (Image image = pdf.SaveAsImage(i, 300, 300))
                {
                    image.Save(fileName, ImageFormat.Png);
                }

                //////////////////Use the following code for netstandard dlls/////////////////////////
                /*
                using (var image = pdf.SaveAsImage(i, PdfImageType.Bitmap))
                {
                    string filename = String.Format(outputFile + i + ".bmp");
                    System.IO.FileStream fileStream = new System.IO.FileStream(filename, System.IO.FileMode.Create, System.IO.FileAccess.Write);
                    
                    image.CopyTo(fileStream);
                    fileStream.Flush();
                }

                */
            }

            pdf.Close();
        }

    }
}

File: VB/03_Images/ConvertAllPagesToPNG.vb
Imports System.Drawing.Imaging
Imports Spire.Pdf

Namespace ConvertAllPagesToPNG
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            Dim file As String = "..\..\..\..\..\..\Data\ToImage.pdf"

            ' Create a new PdfDocument object
            Dim pdf As New PdfDocument()

            ' Load the PDF document from file
            pdf.LoadFromFile(file)

            ' Save each page as a PNG image
            For i As Integer = 0 To pdf.Pages.Count - 1
                ' Generate a file name for the image
                Dim fileName As String = String.Format("ToPNG-img-{0}.png", i)

                ' Save the page as an image in PNG format with a resolution of 300x300 dpi
                Using image As Image = pdf.SaveAsImage(i, 300, 300)
                    image.Save(fileName, ImageFormat.Png)
                End Using
            Next i

            ' Close the PDF document
            pdf.Close()
        End Sub

    End Class
End Namespace
